espnet/nets/pytorch_backend/conformer/convolution.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Johns Hopkins University (Shinji Watanabe)
+#                Northwestern Polytechnical University (Pengcheng Guo)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""ConvolutionBlock layer definition."""
+
+from torch import nn
+
+
+class ConvolutionBlock(nn.Module):
+    """ConvolutionBlock layer.
+
+    :param int channels: channels of cnn
+    :param int kernel_size: kernerl size of cnn
+
+    """
+
+    def __init__(
+        self,
+        channels,
+        kernel_size,
+        stride=1,
+        padding=0,
+        dilation=1,
+        bias=False,
+        activation=nn.ReLU(),
+    ):
+        """Construct an ConvolutionBlock object."""
+        super(ConvolutionBlock, self).__init__()
+        self.pad_left = nn.ConstantPad1d((kernel_size - 1, 0), 0)
```

Comment:
Why is this padding asymmetric?

Reply:
In the Gated CNN LM paper (https://arxiv.org/pdf/1612.08083.pdf), they shift the convolutional inputs to prevent the kernels from seeing the future context. I think we shouldn't follow this setup since we can see future information in the speech scenario. I will modify this, thanks so much.

Reply:
For streaming experiments in future research, how about making it a option?
We can use symmetric padding as default in the offline experiments.
But it might be redundant because Transformer encoder generally adopts blockwise encoding instead of framewise encoding.